src/engines/TimeToLive.js

```diff
@@ -0,0 +1,91 @@
+import DefaultHashTable from '../dataStructure/HashTable';
+import DoublyLinkedList from '../dataStructure/DoublyLinkedList';
+import * as hashTableProp from '../hashTableSymbol';
+
+function TimeToLive({ HashTable = DefaultHashTable, defaultTTL } = {}) {
+  const store = new HashTable();
+  const timeSeriesIndex = new HashTable();
+  const timeIndexInterval = 5 * 60 * 1000; // milliseconds.
+
+  let lastRunGC = Date.now();
+
+  this.add = (key, value, ttl = defaultTTL) => {
+    if (!ttl)
```

Comment:
External values need to be evaluated hard. In this case is should be a positive integer

Reply:
implemented the check